nominal/_utils.py

```diff
@@ -44,6 +47,47 @@ def reader_writer() -> Iterator[tuple[BinaryIO, BinaryIO]]:
         r.close()
 
 
+def warn_on_deprecated_argument(
+    argument_name: str, warning_message: str
+) -> Callable[[Callable[Param, T]], Callable[Param, T]]:
+    """Decorator to warn when a deprecated argument is used.
+
+    Args:
+        argument_name: Name of the argument that is deprecated
+        warning_message: Custom warning message to display when the deprecated argument is used
+
+    Returns:
+        A decorator function that warns when the deprecated argument is used
+    """
+
+    def decorator(func: Callable[Param, T]) -> Callable[Param, T]:
+        sig = inspect.signature(func)
+        param_names = list(sig.parameters.keys())
+
+        @wraps(func)
+        def wrapper(*args: Param.args, **kwargs: Param.kwargs) -> T:
+            if argument_name in kwargs:
+                warnings.warn(warning_message, UserWarning, stacklevel=2)
+                filtered_kwargs = kwargs.copy()
+                filtered_kwargs.pop(argument_name)
+                return func(*args, **cast(Param.kwargs, filtered_kwargs))
```

Comment:
technically this cast is wrong, but I'm not sure it's important...either way, I think I'd just not cast and instead do the specific ignore?